### ISSUE TITLE
Browser support for @truffle/hdwallet-provider 

### ISSUE THREAD

**Issue by mitar** [trufflesuite/truffle#2374]
## Issue

It looks like `@truffle/hdwallet-provider` does not wok in the browser.

## Steps to Reproduce

Try doing `const HDWalletProvider = require('@truffle/hdwallet-provider');` in the browser, in an app bundled with Webpack.

## Expected Behavior

That import will work.

## Actual Results

When I try to do so, I get the error:

```
Uncaught TypeError: Cannot read property 'toString' of undefined
```

**Comment by mbiegert** [comment]
I would like to second this issue. Or are we missing something here?

I get three module not found errors: `child_process`, `fs` and `net`, which is obviously understandable since we are not running on node but in the browser.

**Comment by CruzMolina** [comment]
Hey @mbiegert & @mitar. Currently `@truffle/hdwallet-provider` is not [bundled](https://github.com/trufflesuite/truffle/blob/develop/packages/hdwallet-provider/webpack/webpack.config.js#L9) for browser usage.

You might want to give [`react-native-truffle-hdwallet-provider`](https://github.com/bpeters/react-native-truffle-hdwallet-provider) a try.

**Comment by mitar** [comment]
It looks like that one requires react-native for crypto, which is not the web platform. But yes, something similar where crypto is provided by WebCrypto should be made.

**Comment by mbiegert** [comment]
> It looks like that one requires react-native for crypto, which is not the web platform. But yes, something similar where crypto is provided by WebCrypto should be made.

Is there something we could possibly do to facilitate that?
Maybe it's worth stating explicitly in the readme that this is not browser compatible. The requirements list `node`, but still I was thinking that it might work at first.

If I get it correctly, one would need to drop dependencies on node only packages (child_process, fs, crypto) and come up with a replacement/use `WebCrypto`.

**Comment by mitar** [comment]
So for many node-only packages you can get browser-compatible replacements which provide the same API. The question is more if some features are needed at all in the browser, like, why is `child_process` even needed?

**Comment by haltman-at** [comment]
So, `@truffle/hdwallet-provider` itself does not rely on `child_process`; rather, some of its dependencies do.  As such, there's not much we can do about this at the moment, but we'll keep this open with the hope that we can fix it in the future.

**Comment by mitar** [comment]
One can always mock them with dummy functions. If `hdwallet-provider` never really invokes that code path, this will work out.

**Comment by gnidan** [comment]
Sounds like that's worth a shot @mitar!

**Comment by MrRefactoring** [comment]
With the release of version 1.0.19, I got the opportunity to build a package for the browser, but there was a problem with the `debug` package that `hdwallet-provider` uses, do we really need it?

**Comment by gnidan** [comment]
I'd be fine removing it, although it's common throughout the monorepo to use that module. I worry it might come back to bite someone later if it gets accidentally put there again.

If we do remove it, probably best to leave a note at the top of the code with a link to [@MrRefactoring's comment](https://github.com/trufflesuite/truffle/issues/2374#issuecomment-536109086), to provide the reasoning.

**Comment by eggplantzzz** [comment]
Closing as it was released in Truffle 5.0.39. Here is the PR https://github.com/trufflesuite/truffle/pull/2427
